src/sidebar/components/sidebar-content.js

```diff
@@ -252,8 +250,14 @@ function SidebarContentController(
   // Re-fetch annotations when focused group, logged-in user or connected frames
   // change.
   $scope.$watch(
-    () => [groups.focused().id, store.profile().userid, ...store.searchUris()],
-    ([currentGroupId], [prevGroupId]) => {
+    () => [groups.focused(), store.profile().userid, ...store.searchUris()],
```

Comment:
I haven't checked to see if there are any unforeseen implications of this change. We would be comparing references instead of id's here but since the focused group is grabbed from the groups list which is static until it's reloaded, I think that should be ok. I might dig a little deeper into that tomorrow. Let me know if you notice/know of any!

Reply:
Update: Didn't find any.

Reply:
There is a subtle issue here. Because the last argument to `$scope.$watch` is `true`, a deep equality check is done for all of the arguments. For a complex object graph that would be inefficient. The deep check is needed for `searchUris` because that can currently return a different reference but with the same content if some non-URL property of the connected documents changes. If `groups.focused()` can return `null` then the previous code was already broken though because it assumed otherwise.

Reply:
Hmmm...well technically it could have been null but that never happened in reality because it was always set before this func began watching. I'm not sure what to do about what you describe above. Any ideas? There are quite a few attrs on groups. Not enough to cause a significant slow down at the moment but I see what you are saying.

Reply:
Perhaps this is one of those leave it for now and address this in a follow-up later? In general I'd like to see if there's a way we could get rid of this watch method all together but that requires a larger discussion and I'm not gonna tackle that in this PR.